src/rendering/renderers/shared/shader/UniformGroup.ts

```diff
@@ -152,15 +154,16 @@ export class UniformGroup<UNIFORMS extends { [key: string]: UniformData } = any>
             uniformData.name = i;
             uniformData.size = uniformData.size ?? 1;
 
-            if (!UNIFORM_TYPES_MAP[uniformData.type])
-            {
-                // eslint-disable-next-line max-len
-                throw new Error(`Uniform type ${uniformData.type} is not supported. Supported uniform types are: ${UNIFORM_TYPES_VALUES.join(', ')}`);
-            }
-
             uniformData.value ??= getDefaultUniformValue(uniformData.type, uniformData.size);
 
-            uniforms[i] = uniformData.value as ExtractUniformObject<UNIFORMS>[keyof UNIFORMS];
+            if (options.cloneValues && (uniformData.value as any).slice)
```

Comment:
Arrays and typedBuffers, this will go for anyone who spawn UniformGroup's with the same default values, I do not know how did you evade this in your project. I believe it belongs here, not app level. We can make it in another PR and discuss, if you are not sure, because there we'll be MANY things like that when I make Geometry PR.

Reply:
lets remove this for now and considor it a seperate PR.
my main reasoning is that cloneValues implies it will clone all the values, but it wont - only the arrays. This would be confusing to users.  I would like us to think about htis a bit more - but I don't want to hold up this PR so lets remove it for now.

We can have the logic moved to when a uniform group is created or potentially extend UniforGroup in the short term.